lib/player_turn.rb

```diff
@@ -0,0 +1,36 @@
+require_relative "coordinates"
+require_relative "null_coordinates"
+
+class PlayerTurn
+  attr_reader :input, :view
+
+  def initialize(input:, view:)
+    @input = input
+    @view = view
+  end
+
+  def get_coordinates
+    until coordinates.valid?
+      get_and_validate_coordinates
+    end
+    coordinates.value
+  end
+
+  private
+
+  def coordinates
+    @coordinates || NullCoordinates.new
+  end
+
+  def get_and_validate_coordinates
```

Comment:
Ahh! This is understandably complex, but the naming and responsibilities are lining up to makes for better comprehension.